src/Smearing.jl

```diff
@@ -130,10 +130,12 @@ function MethfesselPaxton(order::Integer)
     end
 end
 
-# TODO: Marzari-Vanderbilt "cold smearing"
+struct Cold <: SmearingFunction end
+occupation(S::Cold, x) = 2 - erfc(x - 1/√2) + sqrt(2 / typeof(x)(π)) * exp(-x^2 + √2x - 1/2)
```

Comment:
You have to be a bit more careful here: x - 1/sqrt(2) will compute 1/sqrt(2) in Float64, for a loss of precision if you're computing in higher precision. Do typeof(2). Same for 1/2. Yeah it's annoying: there's a macro to do it automatically (ChangePrecision.jl) but here it's just simpler to do it.

Actually for this function and the others, it'd be nicer to type the argument (`occupation(S::Cold, x::T)`) and then do T(2), not sure why we didn't do it before.

Reply:
Also I worry reading this whether √2x is √2 x or √(2x), can you add parentheses on this?

Reply:
> Actually for this function and the others, it'd be nicer to type the argument (occupation(S::Cold, x::T)) and then do T(2), not sure why we didn't do it before.

agree.


> Also I worry reading this whether √2x is √2 x or √(2x), can you add parentheses on this?


Or just use the sqrt function :smile:

Reply:
I can go through all of the functions so that where necessary the signatures are `occupation(S::SmearingFunction, x::T) where T` and `entropy(S::Smearingfunction, x::T) where T`.
This way, I can try to clean up all of the `typeof(x)(pi)` etc.

I'll also just use the `sqrt` function 😅 .